src/Features/Core/Portable/Workspace/BackgroundCompiler.cs

```diff
@@ -98,6 +100,11 @@ private void Rebuild(Solution solution, ProjectId initialProject = null)
                 // build the current compilations without rebuilding the entire DeclarationTable
                 CancelBuild(releasePreviousCompilations: false);
 
+                if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(solution.Options) == BackgroundAnalysisScope.ActiveFile)
+                {
+                    return;
+                }
```

Comment:
this needs commenting as to why this is ok.  for example, why is this in the lock, why does it happen ater 'cancel build' is called.  what causes the build to rebuild properly if the bgscope is changed by the user?